test/CadlRanchProjects.Tests/type-property-nullable.cs

```diff
@@ -285,5 +289,175 @@ public Task Type_Property_Nullable_CollectionsString_patchNull() => Test(async (
             var response = await new NullableClient(host, null).GetCollectionsStringClient().PatchNullAsync(RequestContent.Create(value));
             Assert.AreEqual(204, response.Status);
         });
+
+        [Test]
+        public void RequiredNullableAreNotWriteable()
+        {
+            var requiredStringList = TypeAsserts.HasProperty(typeof(StringProperty), nameof(StringProperty.NullableProperty), BindingFlags.Public | BindingFlags.Instance);
+
+            Assert.Null(requiredStringList.SetMethod);
+        }
+
+        [Test]
+        public void RequiredNullableAreNotSetable()
+        {
+            var requiredStringList = TypeAsserts.HasProperty(typeof(StringProperty), nameof(StringProperty.NullableProperty), BindingFlags.Public | BindingFlags.Instance);
+
+            Assert.Null(requiredStringList.SetMethod);
+        }
+
+        [Test]
+        public void NotRequiredNullableListsSerializedAsNull()
+        {
+            var inputModel = new CollectionsStringProperty();
+            inputModel.NullableProperty = null;
+
+            var element = JsonAsserts.AssertWireSerializes(inputModel);
+            Assert.AreEqual(JsonValueKind.Null, element.GetProperty("nullableProperty").ValueKind);
+        }
+
+        [Test]
+        public void NotRequiredNullableListsSerializedEmptyWhenCleared()
+        {
+            var inputModel = new CollectionsStringProperty("required", new List<string> {"1", "2"});
+            inputModel.NullableProperty.Clear();
+
+            var element = JsonAsserts.AssertWireSerializes(inputModel);
+            Assert.AreEqual(JsonValueKind.Array, element.GetProperty("nullableProperty").ValueKind);
+        }
+
+        [Test]
+        public void NotRequiredNullablePropertiesSerializeWhenSet()
+        {
+            var inputModel = new CollectionsStringProperty();
+            inputModel.RequiredProperty = "required";
+            inputModel.NullableProperty = new List<string> { "1", "2" };
+
+            var element = JsonAsserts.AssertWireSerializes(inputModel);
+            Assert.AreEqual("2", element.GetProperty("nullableProperty")[1].GetString());
+        }
+
+        [Test]
+        public void NotRequiredNullablePropertiesDeserializedWithNulls()
+        {
+            var model = CollectionsStringProperty.DeserializeCollectionsStringProperty(JsonDocument.Parse("{\"NonRequiredNullableString\": null}").RootElement);
+            Assert.Null(model.NullableProperty);
```

Comment:
The json name of this `NullableProperty` is `nullableProperty`.
We should not deserialize a irrelevant json here.

Reply:
fixed